src/Storages/ExternalDataSourceConfiguration.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <Parsers/ASTIdentifier.h>
+#include <Interpreters/Context.h>
+#include <Poco/Util/AbstractConfiguration.h>
+
+
+namespace DB
+{
+
+struct ExternalDataSourceConfiguration
+{
+    String host;
+    UInt16 port = 0;
+    String username;
+    String password;
+    String database;
+    String table;
+    String schema;
+
+    std::vector<std::pair<String, UInt16>> addresses; /// Failover replicas.
```

Comment:
I think std::pair is also ok. It is used in too many places and also in base directory, but base directory does not have access to definitions in dbms. It will require to copy paste it there also.